src/analyzer/schema-generator.ts

```diff
@@ -271,6 +279,14 @@ async function generateDataModelsForFiles({
 }: GenerateDataModelsOptions): Promise<{ dataModels: PartialDataModel[]; objectModels: PartialObjectModel[] }> {
     const dataModels: PartialDataModel[] = [];
     let modelNameCounter = 1;
+
+    if (filePaths.length === 0) {
+        return {
+            dataModels: [],
+            objectModels: []
+        };
+    }
+
```

Comment:
same here...
what this code defend agains?

Reply:
if `filePaths` are empty, the for loop will not run and everything will be fine

Reply:
if `filePaths` are empty, the for loop will not run and everything will be fine

Reply:
also, if it was ok, it still was wrong, you need to pass original `objectModels`.
read the code :)